docs/developer-docs/smart-contracts/advanced-features/management-canister.mdx

```diff
@@ -84,14 +84,40 @@ async fn install_code(canister_id: CanisterId, wasm_module: Vec<u8>) {
 <AdornedTab value={"typescript"} label="TypeScript" endAdornment={<BetaChip />}>
 
 ```typescript
-@update([], CreateCanisterResult)
-async executeCreateCanister(): Promise<CreateCanisterResult> {
-    const createCanisterResult = await createCanister();
+import { call, IDL, Principal, query, update } from 'azle';
+import {
+    CanisterInfoArgs,
+    CreateCanisterArgs,
+    CreateCanisterResult,
+} from 'azle/canisters/management';
+
+type State = {
+    createdCanisterId: Principal;
+};
+
+let state: State = {
+    createdCanisterId: Principal.fromText('aaaaa-aa')
+};
 
-    state.createdCanisterId = createCanisterResult.canister_id;
+export default class {
+    @update([], CreateCanisterResult)
+    async executeCreateCanister(): Promise<CreateCanisterResult> {
+        const createCanisterResult = await createCanister();
 
-    return createCanisterResult;
+        state.createdCanisterId = createCanisterResult.canister_id;
+
+        return createCanisterResult;
+    }
+}
+async function createCanister(): Promise<CreateCanisterResult> {
```

Comment:
I would get rid of this function and just paste its body into `executeCreateCanister`, and remove the `state` mutation and variable, those were for getting our tests to pass.

Up until recently our "examples" have really been tests. We've just divided the true show horse examples from the examples that are meant to test.